M2/Macaulay2/packages/ResolutionsOfStanleyReisnerRings.m2

```diff
@@ -0,0 +1,589 @@
+   for k in NonGradedKSOP do (
+       KSOP = append(KSOP,psi(k));
+   );
+
+   phi       := map(S, R, matrix { KSOP });
+   MB        := pushForward(phi, S^1/I);
+   return(MB);
+);
+
+colorfulPresentation (List) := F -> (
+   D         := simplicialComplex(F);
+   B0        := barycentricSubdivision(D);
+   STrivial  := ring(B0);
+   B         := faceDelete(STrivial_0,B0);
+   r         := dim(D)+1;
+   R         := coefficientRing(STrivial)[vars(1..r), Degrees => splice{1..r}];
+
+   -----Constructs graded balanced polynomial ring-----
+   seqOfDegrees := ();
+   for i from 1 to #gens(STrivial) do (
+       seqOfDegrees = join(seqOfDegrees,(#faces(i-2,D, useFaceClass=>true):i-1));
+   );
+
+   ListOfDegs    := toList(seqOfDegrees);
+   S             := newRing(STrivial, Degrees => ListOfDegs);
+
+   -----psi induces the grading on all ring related elements-----
+   psi           := map(S,STrivial);
+   I             := psi(ideal(B));
+   NonGradedKSOP := colorfulSOP(B,#faces(0,D,useFaceClass=>true));
+   KSOP          := {};
+   for k in NonGradedKSOP do (
+       KSOP = append(KSOP,psi(k));
+   );
+
+   phi       := map(S, R, matrix { KSOP });
+   MB        := pushForward(phi, S^1/I);
+   return(MB);
+);
+---------------------------------------
+--Universal System of Parameters (USOP)
+---------------------------------------
+
+universalSOP = method(TypicalValue => List)
+universalSOP (SimplicialComplex) := D -> (
+   Theta    := {};
+   for i from 0 to dim(D) do ( 
+     elts   := flatten(entries(faces(i,D)));
+     theta  := sum for j in elts list j;
+     Theta   = append(Theta,theta);
+   );
+   return Theta;
+);
+
+universalSOP (List) := F -> (
+   D        := simplicialComplex(F);
+   Theta    := {};
+   for i from 0 to dim(D) do ( 
+     elts   := flatten(entries(faces(i,D)));
+     theta  := sum for j in elts list j;
+     Theta   = append(Theta,theta);
+   );
+   return Theta;
+);
+
+
+----------------------------------------
+--Standard Main Function
+----------------------------------------
+
+universalPresentation = method(TypicalValue => Module)
+universalPresentation (List) := F -> (
+   D        := simplicialComplex(F);
+   S        := ring(D);
+   r        := dim(D)+1;
+   USOPRing := coefficientRing(ring(D))[vars(1..r), Degrees => splice{1..r}];
+   phi      := map(S, USOPRing, matrix {universalSOP(D)});
+   MD       := pushForward(phi,S^1/ideal(D));
+   return(MD);
+);
+
+universalPresentation (SimplicialComplex) := D -> (
+   S        := ring(D);
+   r        := dim(D)+1;
+   USOPRing := coefficientRing(ring(D))[vars(1..r), Degrees => splice{1..r}];
+   phi      := map(S, USOPRing, matrix {universalSOP(D)});
+   MD       := pushForward(phi,S^1/ideal(D));
+   return(MD);
+);
+----------------------------------------
+--Sum Betti numbers of resolutions over parameter rings
+----------------------------------------
+
+sumBetti = method(TypicalValue => ZZ)
+sumBetti (Module) := M -> (
+   return(rank(sum(res(M))));
```

Comment:
Another way would be `sum values betti res M`.  But it's so trivial, it doesn't deserve a new function just for that.

Reply:
I agree it is definitely trivial for those who know m2 or have basic programing skills to figure out, but since I needed it for this package, I thought I would include for those mathematicians who are less literate to simply call upon. That was my reasoning, but I can remove it if you still think it's too trivial.

Reply:
ok